src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -486,12 +486,18 @@ public int Length
                         // Avoid possible infinite capacity growth.  See https://github.com/dotnet/coreclr/pull/16926
                         int capacityToPreserve = Math.Min(Capacity, Math.Max(Length * 6 / 5, m_ChunkChars.Length));
                         int newLen = capacityToPreserve - chunk.m_ChunkOffset;
-                        char[] newArray = new char[newLen];
-
-                        Debug.Assert(newLen > chunk.m_ChunkChars.Length, "The new chunk should be larger than the one it is replacing.");
-                        Array.Copy(chunk.m_ChunkChars, 0, newArray, 0, chunk.m_ChunkLength);
+                        if (newLen > chunk.m_ChunkChars.Length)
+                        {
+                            char[] newArray = new char[newLen];
+                            Array.Copy(chunk.m_ChunkChars, 0, newArray, 0, chunk.m_ChunkLength);
+                            m_ChunkChars = newArray;
+                        }
+                        else
+                        {
+                            Debug.Assert(newLen == chunk.m_ChunkChars.Length, "The new chunk should be larger or equal to the one it is replacing.");
+                            m_ChunkChars = chunk.m_ChunkChars;
```

Comment:
```
// Special case where the capacity we want to keep corresponds exactly to the size of the content.
// Just take ownership of the array.
```

Not sure the assert is worth having now.

Reply:
Can you check there's a test that goes down this path?

Reply:
I'll move this comment into the else block